File: Archive#02-201-300/260-Single Number III/leetcode260.rb
# @param {Integer[]} nums
# @return {Integer[]}
def single_number(nums)
  hoge = nums.reduce(:^)
  hoge &= -hoge
  a = nums.select { |e| e & hoge == 0 }.reduce(:^)
  b = nums.select { |e| e & hoge != 0 }.reduce(:^)
  [a, b]
end

File: Archive#02-201-300/260-Single Number III/leetcode260.py
from functools import reduce

class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        flag = reduce(lambda a, b: a ^ b, nums)
        flag &= -flag
        a, b = 0, 0
        for e in nums:
            if e & flag: a ^= e
            else: b ^= e
        return [a, b]